File: KvizHub.QuizService/Models/Entities/Category.cs
using System.ComponentModel.DataAnnotations;

namespace KvizHub.QuizService.Models.Entities
{
    public class Category
    {
        [Key]
        public int Id { get; set; }

        [Required, MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public ICollection<Quiz> Quizzes { get; set; } = new List<Quiz>();
    }
}

File: kvizhub-frontend/src/models/Category.js
/**
 * Category model – represents a quiz category (e.g. Programiranje, Istorija)
 */
export class Category {
  constructor({ id, name, description, quizCount }) {
    this.id = id;
    this.name = name || '';
    this.description = description || '';
    this.quizCount = quizCount || 0;
  }
}

export default Category;
